hibernate-core/src/test/java/org/hibernate/orm/test/annotations/DatabaseTimestampsColumnTest.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.orm.test.annotations;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.Id;
+import org.hibernate.annotations.GenerationTime;
+import org.hibernate.annotations.NaturalId;
+import org.hibernate.annotations.ValueGenerationType;
+import org.hibernate.testing.orm.junit.DialectFeatureChecks;
+import org.hibernate.testing.orm.junit.EntityManagerFactoryScope;
+import org.hibernate.testing.orm.junit.Jpa;
+import org.hibernate.testing.orm.junit.RequiresDialectFeature;
+import org.hibernate.tuple.AnnotationValueGeneration;
+import org.hibernate.tuple.GenerationTiming;
+import org.hibernate.tuple.ValueGenerator;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.util.Date;
+
+@RequiresDialectFeature( feature = DialectFeatureChecks.UsesStandardCurrentTimestampFunction.class )
+@Jpa(annotatedClasses = DatabaseTimestampsColumnTest.Person.class)
+public class DatabaseTimestampsColumnTest {
+
+    @Entity(name = "Person")
+    public class Person {
+
+        @Id
+        @GeneratedValue
+        private Long id;
+
+        @NaturalId(mutable = true)
+        private String name;
+
+        @Column(nullable = false)
+        @Timestamp(GenerationTime.INSERT)
+        private Date creationDate;
+
+        @Column(nullable = true)
+        @Timestamp(GenerationTime.UPDATE)
+        private Date editionDate;
+
+        @Column(nullable = false, name="version")
+        @Timestamp(GenerationTime.ALWAYS)
+        private Date timestamp;
+
+        public String getName() {
+            return name;
+        }
+
+        public void setName(String name) {
+            this.name = name;
+        }
+
+        public Date getCreationDate() {
+            return creationDate;
+        }
+
+        public Date getEditionDate() {
+            return editionDate;
+        }
+
+        public Date getTimestamp() {
+            return timestamp;
+        }
+    }
+
+    @ValueGenerationType(generatedBy = TimestampValueGeneration.class)
+    @Retention(RetentionPolicy.RUNTIME)
+    public @interface Timestamp { GenerationTime value(); }
```

Comment:
I guess you could use `@CurrentTimestamp` instead of defining a custom annotation?

Reply:
Well of course. But it's a test.